<launch>
    <!--MH_01_easy MH_02_easy MH_03_medium MH_04_difficult MH_05_difficult V1_01_easy V1_02_medium V1_03_difficult V2_01_easy V2_02_medium V2_03_difficult  -->
    <arg name="dataset_name" default = "V1_03_difficult" />
    <!-- change to your euroc dataset path-->
    <arg name="dataset_bag_path" default = "/home/tony-ws/bag/ijrr_euroc_mav_dataset" />

    <node pkg="rosbag" type="play" name="player" output="log" 
    args="$(arg dataset_bag_path)/$(arg dataset_name)/$(arg dataset_name).bag " />

    <node name="feature_tracker" pkg="feature_tracker" type="feature_tracker" output="screen">
        <!-- camera calibration file. Be care about the format -->
        <param name="calib_dir" type="string" value="$(find feature_tracker)/config/" />
        <param name="cam_name0" type="string" value="euroc0_camera_calib_mei.yaml" />
        <!-- max feature number in feature tracking -->
        <param name="max_cnt" type="int" value="200" />
        <!-- min distance between two features -->
        <param name="min_dist" type="int" value="20" />
        <!-- pub freq -->
        <param name="freq" type="int" value="1" />
         <!-- ransac threshold -->
        <param name="F_threshold" type="double" value="1.0" />
        <!-- if image is too dark or light, trun on equalize to find enough features -->
        <param name="show_track" type="bool" value="true" />
        <!-- if image is too dark or light, trun on equalize to find enough features -->
        <param name="equalize" type="bool" value="false" />
        <!-- if using fisheye, trun on it. A circle mask will be loaded to remove edge noisy points -->
        <param name="fisheye" type="bool" value="false" />
        <!-- remap to your imu and image topic-->
        <remap from="~imu" to="/imu0" />
        <remap from="~raw_image" to="/cam0/image_raw" />
    </node>
</launch>
